records/generator.go

```diff
@@ -271,12 +271,15 @@ func (rg *RecordGenerator) InsertState(sj StateJSON, domain string, ns string,
 
 	// creates a map with slave IP addresses (IPv4)
 	rg.SlaveIPs = make(map[string]string)
+	rg.SRVs = make(rrs)
+	rg.As = make(rrs)
+
 	for _, slave := range sj.Slaves {
-		rg.SlaveIPs[slave.ID] = sanitizedSlaveAddress(slave.Hostname, spec)
+		ssa := sanitizedSlaveAddress(slave.Hostname, spec)
```

Comment:
@jdef does that affect rg.SlaveIPs[slave.ID] as well? or is it okay for rg.SlaveIPs to contain invalid IPs? I.e. should I only check for the part where I add an A record for the slave.domain name or also for the rg.SlaveIPs map?